nexus/types/src/deployment.rs

```diff
@@ -754,6 +755,7 @@ impl From<BlueprintZoneConfig> for OmicronZoneConfig {
             id: z.id,
```

Comment:
I know I'm being overly cautious here, but what do you think about adding:

```rust
/// This match helps catch cases where we've added fields to `BlueprintZoneConfig` but forgotten to add them to `OmicronZoneConfig`.  If you get a compile error here, consider whether `OmicronZoneConfig` needs an update for your changes.
let BlueprintZoneConfig {
    disposition: _disposition,
    id,
    filesystem_pool,
    zone_type
} = z;
```

and then use those bindings below, mainly so that if/when we add `image_source` to the blueprint, we won't forget to update this.